prestoadmin/file.py

```diff
@@ -32,19 +32,19 @@
 
 @task
 @requires_config(StandaloneConfig)
-def run(script, remote_dir='/tmp'):
+def run(script, params=''):
```

Comment:
This requires params encoding in shell-specific way (the way how `sudo()` method works).
It works great for simple params, but will be problematic as soon as there is any whitespace or quotation marks.
Better API would be to accept params as a list (`params=()` as the default)

Reply:
@findepi, so if I understand I should get the params as a list and then run `" ".join(params)`?

Reply:
You need proper escaping of the params. For example with script file:
```bash
#!/bin/bash
exec mkdir "$@"
```
invoking
```
run("mkdir-script", params=("-p", "/tmp/name with space/name with ' apostrophe/xxx"))
```
should result the following dir structure being created
```
/tmp
    /name with space
        /name with ' apostrophe
            /xxx
```

Notes:
- i don't know fabric good enough to know whether it provides some proper way of doing that
- you need to verify it doesn't get broken by ssh + sudo piping. (_If this is a  case_, you'd need to upload the request script _and_ a wrapper script that calls the first with required params.)
  - please add a test to `tests/product/test_file.py` too